exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DropTableHandler.java

```diff
@@ -56,7 +57,7 @@ public DropTableHandler(SqlHandlerConfig config) {
   @Override
   public PhysicalPlan getPlan(SqlNode sqlNode) throws ValidationException, RelConversionException, IOException {
     SqlDropTable dropTableNode = ((SqlDropTable) sqlNode);
-    String originalTableName = dropTableNode.getName();
+    String originalTableName = FileSelection.removeLeadingSlash(dropTableNode.getName());
```

Comment:
Maybe instead of removing slashes we should disallow creating such tables?

Reply:
Actually, this may appear counter-intuitive for users especially when Drill supports `SELECT` with leading slash (see [DRILL-1130](https://issues.apache.org/jira/browse/DRILL-1130)). When a table is created and its name does start with '/' it will be created within workspace, i.e. will be rooted from the workspace.
You could think of it as if you performed `touch` (on Unix[-like] OS) to create a file: `touch /${filename}` and `touch ${filename}` should create a file in the same directory when working directory (workspace) is root ('/').

Reply:
Sounds fair, thanks for the explanation.